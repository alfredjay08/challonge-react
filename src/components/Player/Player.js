import React from "react";
import { PlayerContainer, Column } from "../StyledComponents/StyledComponents";

const Player = ({
  playerInfo,
  playerScore,
  winner = false,
  hovered,
  unhovered,
  highlighted = false,
}) => {
  const imgStyle = {
    width: "20px",
    borderRadius: "50%",
  };

  return (
    <PlayerContainer
      onMouseOver={() => {
        if (!playerInfo) return;

        hovered(playerInfo._id);
      }}
      onMouseOut={unhovered}
      highlighted={highlighted}
    >
      <Column>
        <img
          src="/uploads/ninja-yellow.jpg"
          alt="Player Icon"
          style={imgStyle}
        />
      </Column>
      <Column grow>{playerInfo && playerInfo.name}</Column>
      <Column score winner={winner}>
        <span>{playerInfo && playerScore}</span>
      </Column>
    </PlayerContainer>
  );
};

export default Player;
