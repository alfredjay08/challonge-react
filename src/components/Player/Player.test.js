import React from "react";
import renderer from "react-test-renderer";

import Player from "./Player";

it("renders when there are no props", () => {
  const tree = renderer.create(<Player />).toJSON();

  expect(tree).toMatchSnapshot();
});

it("renders when there are incorrect props", () => {
  const tree = renderer
    .create(
      <Player
        wrongProp={10}
        wrongStyle={{ style: "background" }}
        wrongPlayer={{ playerName: "Alicred" }}
        wrongPlayer2={{ playerName: "Derkila" }}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
