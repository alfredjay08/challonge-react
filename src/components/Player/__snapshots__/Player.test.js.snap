// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there are incorrect props 1`] = `
<div
  className="sc-jSUZER hFnVAg"
  onMouseOver={[Function]}
>
  <div
    className="sc-gKPRtg jAzlAk"
  >
    <img
      alt="Player Icon"
      src="/uploads/ninja-yellow.jpg"
      style={
        Object {
          "borderRadius": "50%",
          "width": "20px",
        }
      }
    />
  </div>
  <div
    className="sc-gKPRtg fOrJSf"
  />
  <div
    className="sc-gKPRtg jLsyYQ"
  >
    <span />
  </div>
</div>
`;

exports[`renders when there are no props 1`] = `
<div
  className="sc-jSUZER hFnVAg"
  onMouseOver={[Function]}
>
  <div
    className="sc-gKPRtg jAzlAk"
  >
    <img
      alt="Player Icon"
      src="/uploads/ninja-yellow.jpg"
      style={
        Object {
          "borderRadius": "50%",
          "width": "20px",
        }
      }
    />
  </div>
  <div
    className="sc-gKPRtg fOrJSf"
  />
  <div
    className="sc-gKPRtg jLsyYQ"
  >
    <span />
  </div>
</div>
`;
